tests/unit/bootstrap.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * Give Unit Tests Bootstrap
+ *
+ * @unreleased
+ */
+class ADDON_CONSTANT_Unit_Tests_Bootstrap {
+
+	/** @var \Give_Unit_Tests_Bootstrap instance */
+	protected static $instance = NULL;
+
+	/** @var string directory where wordpress-tests-lib    is installed */
+	public $wp_tests_dir;
+
+	/** @var string testing directory */
+	public $tests_dir;
+
+	/** @var string plugin directory */
+	public $plugin_dir;
+
+	/** @var string give directory */
+	public $give_dir;
+
+	/**
+	 * Give_RD_Unit_Tests_Bootstrap constructor.
+	 *
+	 * Setup the unit testing environment
+	 */
+	public function __construct() {
+
+		ini_set( 'display_errors', 'on' );
+		error_reporting( E_ALL );
+
+		// Prevent give_die() from stopping tests.
+		if ( ! defined( 'GIVE_UNIT_TESTS' ) ) {
+			define( 'GIVE_UNIT_TESTS', TRUE );
+		}
+
+		$this->tests_dir    = dirname( __FILE__ );
+		$this->plugin_dir   = dirname( $this->tests_dir );
+		$this->give_dir     = dirname( dirname( $this->plugin_dir ) ) . '/give';
+		$this->wp_tests_dir = getenv( 'WP_TESTS_DIR' ) ? getenv( 'WP_TESTS_DIR' ) : '/tmp/wordpress-tests-lib';
+
+		// Load test function so tests_add_filter() is available
+		require_once( "{$this->wp_tests_dir}/includes/functions.php" );
+
+		// Run bootstrap.php in this file only.
+		$GLOBALS['manual_bootstrap'] = FALSE;
+
+		// Load Give
+        if( file_exists( "{$this->give_dir}/tests/unit/bootstrap.php" ) ) {
+            require_once("{$this->give_dir}/tests/unit/bootstrap.php");
+        } else {
+            require_once("{$this->wp_tests_dir}/give/tests/unit/bootstrap.php");
+        }
+
+		// Uninstall plugin.
+		tests_add_filter( 'plugins_loaded', array( $this, 'uninstall_plugin' ), 0 );
+
+		// Load plugin
+		tests_add_filter( 'muplugins_loaded', array( $this, 'load_plugin' ) );
+
+
+		// Install plugin
+		tests_add_filter( 'init', array( $this, 'install_plugin' ) );
+
+		// Load the WP testing environment
+		require_once( "{$this->wp_tests_dir}/includes/bootstrap.php" );
+
+		// Load Give testing framework
+		$this->includes();
+	}
+
+	/**
+	 * Load recurring itself.
+	 */
+	public function load_plugin() {
+		require_once( dirname($this->plugin_dir) . "/ADDON_ID.php" );
+	}
+
+	/**
+	 * Uninstall plugin
+	 */
+	public function uninstall_plugin() {
+		give_update_option( 'uninstall_on_delete', 'enabled' );
+
+		// WP_UNINSTALL_PLUGIN is already define while uninstalling give.
+//		include( $this->plugin_dir . '/uninstall.php' );
```

Comment:
So do the commented out sections need to be un-commented in order to work? Why not just un-comment them now?

Reply:
My work in GiveWP core suggests that we can now just remove that uninstall hook all together. I'll go ahead and do that now.

Reply:
Cool! Same question for the other commented out sections.

Reply:
As for the other commented out sections, those are placeholders showing where to load custom helpers classes.